query-planner-js/src/buildPlan.ts

```diff
@@ -3649,7 +3657,7 @@ function computeGroupsForTree(
     context: initialContext,
     deferContext: initialDeferContext,
   }];
-  const createdGroups = [ ];
+  const createdGroups: FetchGroup[] = [ ];
```

Comment:
Note to reviewers: I hesitated changing this to a `Set<FetchGroup>`, but this turned out to a bit more changes that felt worth it. But if there is strong feeling about it, it can be changed.

Reply:
A set would certainly be cleaner, but I'll trust your judgement about whether it's worth it to make the change or not.